js/core.js

```diff
@@ -2039,6 +2047,8 @@
                 }
 
                 this._display[context] = {
+                    "isPreformat": (typeof annotation.pre_format === 'string'),
```

Comment:
This seems wrong. You should get the format before this line and in here check if that is string or not.
```javascript
var preformatConfig = typeof annotation.pre_format === 'object') ? ((typeof annotation.pre_format.format === 'string') ? annotation.pre_format : null ) : null;

this._display[context] = {
  "isPreformat": (typeof preformatConfig === 'string'),
  "preformatConfig": preformatConfig,
   //..
}
```